clang/test/SemaSYCL/intel-max-global-work-dim-device.cpp

```diff
@@ -47,14 +61,109 @@ struct TRIFuncObjGood2 {
   operator()() const {}
 };
 
+struct TRIFuncObjGood3 {
+  [[intel::reqd_work_group_size(1, 1, 1)]]
+  [[intel::max_global_work_dim(0)]] void
+  operator()() const {}
+};
+
+struct TRIFuncObjGood4 {
+  [[cl::reqd_work_group_size(1, 1, 1)]]
+  [[intel::max_global_work_dim(0)]] void
+  operator()() const {}
+};
+
+struct TRIFuncObjGood5 {
+  [[intel::max_work_group_size(1, 1, 1)]]
+  [[intel::max_global_work_dim(0)]] void
+  operator()() const {}
+};
+
+struct TRIFuncObjGood6 {
+  [[intel::reqd_work_group_size(4, 1, 1)]]
+  [[intel::max_global_work_dim(3)]] void
+  operator()() const {}
+};
+
+struct TRIFuncObjGood7 {
+  [[cl::reqd_work_group_size(4, 1, 1)]]
+  [[intel::max_global_work_dim(3)]] void
```

Comment:
Test cases caused assertions with existing support of max_global_work_dim attribute. With new design, the test cases generate expected diagnostic.